### DISCUSSION TITLE
First sketch of a "Developer Tutorial"

### DISCUSSION THREAD

**Discussion by EricZinda** [delph-in/docs#109] [Team Posts]
I've put together the first few sections of a "Developer Tutorial" and integrated it into the proposed documentation site.  You can find it [here](https://blog.inductorsoftware.com/docsproto/howto/devhowto/devhowtoOverview/). 

The goal is to help a developer, reasonably proficient in any programming language, understand *one way* to use the Delphin resources to implement a natural language interface to their own software.

The tutorial is missing a few front sections that you folks won't need like "understanding MRS" and "How to build a scope-resolved tree" since that material is more of a known quantity that I can do later.

I'd love feedback from this august body on whether this meets the needs of the target audience, whether it is clear, easy to approach, etc. And whether I have gone too far off the tracks in claiming things about how Delphin or MRS works, or oversimplified it in my simplification effort.

Thanks for taking a look!

**Comment by olzama** [comment]
Forgot that I can't send email from the uw address :)

---------- Forwarded message ---------
From: Olga Zamaraeva ***@***.***>
Date: Tue, Dec 13, 2022 at 4:51 PM
Subject: Re: [delph-in/erdw] First sketch of a "Developer Tutorial" (#8)
To: delph-in/erdw <
***@***.***>, Eric
Zinda ***@***.***>


Hi Eric!

I finally got a chance to take a look!

I am not the target audience, so I probably can't give much feedback, but I
can just give minor comments about the Overview:

0) Maybe links to basic concepts in the Overview section? Similarly to what
you do in the following "Predication contract" section.
1) DELPH-IN (not Delphin, I think)
2) "DELPH-IN languages" -- I'd keep saying "DELPH-IN grammars", or at least
"language for which DELPH-IN grammars exist"

I never really worked with MRS in this way so I cannot really comment on
the contents (as I said), but let me say that I do like what the sections
look like. They give me a feeling at least that they would be very helpful
to me if I needed to do something like that.

Best,
Olga


On Tue, Dec 6, 2022 at 6:52 PM EricZinda ***@***.***> wrote:

> I've put together the first few sections of a "Developer Tutorial" and
> integrated it into the proposed documentation site. You can find it here
> <https://blog.inductorsoftware.com/docsproto/howto/devhowto/devhowtoOverview/>
> .
>
> The goal is to help a developer, reasonably proficient in any programming
> language, understand *one way* to use the Delphin resources to implement
> a natural language interface to their own software.
>
> The tutorial is missing a few front sections that you folks won't need
> like "understanding MRS" and "How to build a scope-resolved tree" since
> that material is more of a known quantity that I can do later.
>
> I'd love feedback from this august body on whether this meets the needs of
> the target audience, whether it is clear, easy to approach, etc. And
> whether I have gone too far off the tracks in claiming things about how
> Delphin or MRS works, or oversimplified it in my simplification effort.
>
> Thanks for taking a look!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/delph-in/teams/erdw/discussions/8>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACTURKTWO3COR6AA2VAAEPTWL54NFANCNFSM6AAAAAASVZYFKI>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Olga

**Comment by EricZinda** [comment]
Thank you @olzama for taking a look! I've fixed up what you suggested, and I will definitely do more links to, and generation of, the overview content. I wanted to start with what I think of as "the new stuff" since it would be more controversial and challenging to write, and then backfill with stuff like "what is an MRS".  I figured the current reviewers knew all the background so it wouldn't hurt their ability to review.

Also, I've done a lot of cleanup, reorganization and added a few sections on "sentence force" and "reporting errors" that I thought were pretty interesting at least :-) It certainly took me a long time to figure them out over these last couple of years.

**Comment by EricZinda** [comment]
OK, I ended up having to write a section describing [MRS](https://blog.inductorsoftware.com/docsproto/howto/devhowto/devhowtoMRS/) and one describing [well-formed MRS Trees](https://blog.inductorsoftware.com/docsproto/howto/devhowto/devhowtoWellFormedTree/) before I could move on with the next section I was hoping to get to.

Love any feedback on those. They really shouldn't be "developer focused" other than the fact that they skew examples in that direction and try to keep things (linguistically) simple. I'll be gone for a couple of weeks, so don't think I'm ghosting you if I don't immediately respond :-)
